src/range.cr

```diff
@@ -125,9 +137,21 @@ struct Range(B, E)
   # # prints: 14 13 12 11 10
   # ```
   def reverse_each : Nil
-    yield @end if !@exclusive && !(@end < @begin)
-    current = @end
-    while @begin < current
+    {% if E == Nil %}
+      {% raise "Can't reverse_each endless range" %}
+    {% end %}
```

Comment:
Why not the same macro check in `#each`?

Reply:
Strange, I though I wrote it... I'll add it

Reply:
Done!